addons/survey/controllers/main.py

```diff
@@ -345,10 +346,19 @@ def _prepare_question_html(self, survey_sudo, answer_sudo, **post):
                     'page_number': page_ids.index(survey_data['question'].id)
                 })
 
+        background_image_url = False
+        if 'question' in survey_data:
+            background_image_url = survey_data['question'].background_image_url
+        elif 'page' in survey_data:
+            background_image_url = survey_data['page'].background_image_url
+        if not background_image_url and survey_sudo.background_image:
+            background_image_url = survey_sudo.background_image_url
```

Comment:
Counter-proposal:

```py
background_image_url = survey_sudo.background_image_url
if 'question' in survey_data:
    background_image_url = survey_data['question'].background_image_url
elif 'page' in survey_data:
    background_image_url = survey_data['page'].background_image_url
```

Since the computed method already handles the fact that there may not be an image configured.
That way we delegate the logic to the model.